recipes/cairo/meson/conanfile.py

```diff
@@ -0,0 +1,250 @@
+        "fPIC": [True, False],
+        "with_freetype": [True, False],
+        "with_fontconfig": [True, False],
+        "with_xlib": [True, False],
+        "with_xlib_xrender": [True, False],
+        "with_xcb": [True, False],
+        "with_glib": [True, False],
+        "with_zlib": [True, False],
+        "with_png": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_freetype": True,
+        "with_fontconfig": True,
+        "with_xlib": True,
+        "with_xlib_xrender": False,
+        "with_xcb": True,
+        "with_glib": True,
+        "with_zlib": True,
+        "with_png": True,
+    }
+
+    exports_sources = "patches/*"
+    generators = "pkg_config"
+
+    _meson = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    @property
+    def _settings_build(self):
+        return getattr(self, "settings_build", self.settings)
+
+    def config_options(self):
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+        if self.settings.os != "Linux":
+            del self.options.with_xlib
+            del self.options.with_xlib_xrender
+            del self.options.with_xcb
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        del self.settings.compiler.cppstd
+        del self.settings.compiler.libcxx
+
+    def requirements(self):
+        self.requires('expat/2.4.1')  # cairo-xml
+        self.requires("pixman/0.40.0")
+        if self.options.get_safe("with_zlib", True):
+            self.requires("zlib/1.2.11")
+        if self.options.get_safe("with_freetype", True):
+            self.requires("freetype/2.11.0")
+        if self.options.get_safe("with_fontconfig", True):
+            self.requires("fontconfig/2.13.93")
+        if self.options.get_safe("with_png", True):
+            self.requires('libpng/1.6.37')
+        if self.options.get_safe("with_glib", True):
+            self.requires("glib/2.70.0")
+        if self.settings.os == "Linux":
+            if self.options.with_xlib or self.options.with_xlib_xrender or self.options.with_xcb:
+                self.requires("xorg/system")
+
+    def build_requirements(self):
+        self.build_requires("meson/0.59.1")
+        self.build_requires("pkgconf/1.7.4")
+
+    @property
+    def _is_msvc(self):
+        return self.settings.compiler == "Visual Studio"
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version],
+                  destination=self._source_subfolder, strip_root=True)
+
+    def _configure_meson(self):
+        yes_no = lambda v: "enabled" if v else "disabled"
+        meson = Meson(self)
+        defs = dict()
+        defs["tests"] = "disabled"
```

Comment:
Does meson accept bot h False and "disabled"? Cool.

Reply:
Oh course, yes_no lambda

Reply:
I don't think it does, right now, only strings